GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1369,10 +1378,10 @@ void DoCommit()
                     {
                         // Save last commit message in settings. This way it can be used in multiple repositories.
                         AppSettings.LastCommitMessage = Message.Text;
-
-                        _commitMessageManager.WriteCommitMessageToFile(Message.Text, CommitMessageType.Normal,
-                                                                       usingCommitTemplate: !string.IsNullOrEmpty(_commitTemplate),
-                                                                       ensureCommitMessageSecondLineEmpty: AppSettings.EnsureCommitMessageSecondLineEmpty);
+                        ThreadHelper.JoinableTaskFactory.RunAsync(
```

Comment:
The file needs to be written before "git commit" is called.
```suggestion
                        ThreadHelper.JoinableTaskFactory.Run(
```